test/dynamo/test_ctx_manager.py

```diff
@@ -905,25 +922,25 @@ def fn(x):
 
         x = torch.rand(2, 3)
         cnts = torch._dynamo.testing.CompileCounter()
-        opt_fn = torch.compile(backend=cnts, fullgraph=False)(fn)
+        opt_fn = torch.compile(backend=cnts, fullgraph=True)(fn)
 
         with torch.no_grad():
             ref = fn(x)
             res = opt_fn(x)
             self.assertTrue(same(ref, res))
-            self.assertEqual(cnts.frame_count, 4)
-            self.assertEqual(cnts.op_count, 4)
+            self.assertEqual(cnts.frame_count, 1)
+            self.assertEqual(cnts.op_count, 9)
 
         with torch.enable_grad():
             ref = fn(x)
             res = opt_fn(x)
             self.assertTrue(same(ref, res))
-            self.assertEqual(cnts.frame_count, 6)
-            self.assertEqual(cnts.op_count, 6)
+            self.assertEqual(cnts.frame_count, 2)
+            self.assertEqual(cnts.op_count, 18)
 
     def test_generic_context_manager_with_graph_break(self):
         def fn(x):
-            with CutomizedCtxManager(True):
+            with CustomizedCtxManager(True):
```

Comment:
@guilhermeleobas can you add some reconstruct tests? that is, we are compiling a function that returns a context manager with fullgraph=True